ci/prow/perf_config.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// data definitions that are used for the config file generation of performance
+// tests cluster maintanence jobs.
+
+package main
+
+import (
+	"fmt"
+	"log"
+)
+
+// generatePerfClusterUpdatePeriodicJobs generates periodic jobs to update clusters
+// that run performance testing benchmarks
+func generatePerfClusterUpdatePeriodicJobs() {
+	for _, repo := range repositories {
+		if repo.EnablePerformanceTests {
+			perfClusterPeriodicJob(
+				"recreate-clusters",
+				recreatePerfClusterPeriodicJobCron,
+				"./test/performance/performance-tests.sh",
+				[]string{"--recreate-clusters"},
+				repo.Name,
+				perfClusterJobSecret(repo.Name),
+			)
+			perfClusterPeriodicJob(
+				"update-clusters",
+				updatePerfClusterPeriodicJobCron,
+				"./test/performance/performance-tests.sh",
+				[]string{"--update-clusters"},
+				repo.Name,
+				perfClusterJobSecret(repo.Name),
+			)
+		}
+	}
+}
+
+// generatePerfClusterPostsubmitJob generates postsubmit job for the
+// repo to reconcile clusters that run performance testing benchmarks.
+func generatePerfClusterPostsubmitJob(repo repositoryData) {
+	perfClusterReconcilePostsubmitJob(
+		"reconcile-clusters",
+		"./test/performance/performance-tests.sh",
+		[]string{"--reconcile-clusters"},
+		repo.Name,
+		perfClusterJobSecret(repo.Name),
+	)
+}
+
+// perfClusterJobSecret returns the secret we need to mount for the given repo
+func perfClusterJobSecret(fullRepoName string) string {
+	// secret that needs to be mounted for the job
+	var secret string
+	switch fullRepoName {
+	case "knative/serving":
+		secret = "performance-test"
+	case "knative/eventing":
+		secret = "eventing-performance-test"
```

Comment:
Dumb question: why can't you use a single secret?

Reply:
The secret was added when we did not have a deep understanding of how Mako worked, but you are right, we can reuse the secret and add the service account as the `OWNER` of all benchmarks, which is much easier to maintain.
Changed to one single secret, and thanks for pointing it out!